test/scratchOrg.ts

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import child_process from 'child_process';
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import {
+  DefaultTreeItem,
+  InputBox,
+  QuickOpenBox,
+  Workbench
+} from 'wdio-vscode-service';
+import {
+  EnvironmentSettings
+} from './environmentSettings';
+import * as utilities from './utilities';
+
+
+const exec = util.promisify(child_process.exec);
+
+export class ScratchOrg {
+  private testSuiteSuffixName: string;
+  private reuseScratchOrg = false;
+  public tempFolderPath: string | undefined = undefined;
+  public projectFolderPath: string | undefined = undefined;
+  private prompt: QuickOpenBox | InputBox | undefined;
+  private scratchOrgAliasName: string | undefined;
+
+  public constructor(testSuiteSuffixName: string, reuseScratchOrg: boolean) {
+    this.testSuiteSuffixName = testSuiteSuffixName;
+    this.reuseScratchOrg = reuseScratchOrg;
+
+    // To have all scratch orgs be reused, uncomment the following line:
+    // this.reuseScratchOrg = true;
+  }
+
+  public get tempProjectName(): string {
+    return 'TempProject-' + this.testSuiteSuffixName;
+  }
+
+  public async setUp(scratchOrgEdition: string = 'Developer'): Promise<void> {
+    await this.setUpTestingEnvironment();
+    await this.createProject(scratchOrgEdition);
+    await this.authorizeDevHub();
```

Comment:
I see how having this step to authorize a dev hub inside of ScratchOrg.setup() makes the call to create a new scratch org for a test super easy.  However, it does seem a bit odd for me to see the auth of the devhub within the scratch org module.   

Have you thought about having some of these as imported from smaller service modules? i.e., TestContext#setupTestingEnvironment(), ProjectService#createProject(), AuthService#authorizeDevHub(), etc.? 

Or, instead of breaking up into smaller modules, maybe renaming this module from scratchOrg.ts to testContext or testSetup.ts?  

const t = new TestSetup();
t.setUpTestingEnvironment();
t.createProject(scratchOrgEdition);
t.authorizeDevHub();

That would feel to me like those methods were in the right place rather than having the scratchOrg module be responsible for doing those things.

Reply:
Done - renamed to TestSetup